src/main/java/spoon/support/compiler/jdt/JDTTreeBuilder.java

```diff
@@ -1754,9 +1757,22 @@ public boolean visit(GuardedPattern guardedPattern, BlockScope scope) {
 
 	@Override
 	public boolean visit(TypePattern anyPattern, BlockScope scope) {
-		CtTypePattern typePattern = factory.Core().createTypePattern();
-		context.enter(typePattern, anyPattern);
-		return true;
+		boolean unnamedPattern = isUnnamedPattern(anyPattern);
+		CtPattern pattern;
+		if (unnamedPattern) {
+			CtTypeReference<?> type = references.getTypeReference(anyPattern.local.binding.type).setImplicit(true);
+			pattern = factory.Core().createUnnamedPattern().setType(type);
+		} else {
+			pattern = factory.Core().createTypePattern();
+		}
+		context.enter(pattern, anyPattern);
+		// when we have an unnamed pattern, we don't want to visit the local variable declaration anymore
+		return !unnamedPattern;
+	}
+
+	// only '_', NOT 'Type _'
+	private boolean isUnnamedPattern(TypePattern pattern) {
+		return pattern.isUnnamed() && pattern.local.type == null && pattern.local.binding != null;
```

Comment:
Cool, so JDT differs from the spec here? :^)

Reply:
They're a bit lazy, they only have `TypePattern` with a variable that can be unnamed (`Type _` as a pattern), but represent the actual unnamed pattern by just not having a type set. `isUnnamed()` in their case refers to the variable, not the kind of pattern, from my understanding.